classes/model/user_default_domain.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/**
+ * Enhanced authentication.
+ *
+ * @author Luke Carrier <luke.carrier@floream.com>
+ * @copyright 2016 Floream Limited
+ */
+
+namespace local_signin\model;
+
+use dml_exception;
+use dml_missing_record_exception;
+use local_signin\interfaces\static_default_domain;
+use local_signin\interfaces\user_domain_interface;
+
+defined('MOODLE_INTERNAL') || die;
+
+class user_default_domain {
+    public $username;
+    public $email;
+    public $domain;
+
+    public function __construct($username=null, $email=null, $domain=null) {
+        $this->username = $username;
+        $this->email    = $email;
+        $this->domain   = $domain;
+    }
+
+    public static function get($input) {
+        global $CFG, $DB;
+
+        $result = new user_default_domain(
+            null,
+            null,
+            parse_url($CFG->wwwroot, PHP_URL_HOST)
+        );
+
+        // Assign a domain finder (either the one stored in $CFG, or the default one).
+        $class = property_exists($CFG, 'local_signin_userdomain') ?
+            $CFG->local_signin_userdomain :
```

Comment:
Can we get this new configuration option be documented in the README?

e.g.

```markdown
## Redirecting users across domains

To facilitate redirecting users to the login process on different domains (e.g. for white labelling), `$CFG->local_signin_userdomain` can optionally be set in `config.php` to the name of a class implementing the `\local_signin\interfaces\user_domain_interface` interface.
```

Reply:
Integrated into the existing documentation.